### PR TITLE
gas price oracle

### PR DESCRIPTION
**Pull request by tallywiesenberg**
I followed along with the template, lmk if anything is missing!

### CODE REVIEW

File: types/GasPriceOracle.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    - value type: `uint256`
+2. timestamp
+    - description: the unix timestamp to calculate on-chain gas price at
+    - value type: `uint256`
+```
+
+
+## Response Type
+
+`GasPriceOracle`'s response type is an unpacked 256 bit value with 18 decimals of precision:
```
Comment (oraclown) [inline types/GasPriceOracle.md:30]: What currency is the gas price?

File: types/GasPriceOracle.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Query Data
+
+Query data is used to form your new Query's unique identifier, or query ID, and it's also included in emitted contract events so Tellor users and reporters can programmatically construct query objects.
+
+To generate the query data for an instance of your new Query type, first UTF-8 encode the parameter values in the order specified above. Then encode those `bytes` with the Query's type string.
+
+For example, to get the query data of an example instance of a `GasPriceOracle` query using Solidity:
+```s
+queryData = abi.encode("GasPriceOracle", abi.encode("1", "1650465649"))
```
Comment (oraclown) [inline types/GasPriceOracle.md:45]: Looks like you're encoding the params as type `string`, when they're `uint256`. I think something like:
```s
queryData = abi.encode("GasPriceOracle", abi.encode(1, 1650465649)
```

File: types/GasPriceOracle.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+
+## Example
+A working example mapping of all the various inputs and parameters to a valid queryID. 
+
+The queryData for mainnet ethereum at unix timestamp `1650465649`:
+
+```s
+queryData = abi.encode("GasPriceOracle", abi.encode("1", "1650465649"))
```
Comment (oraclown) [inline types/GasPriceOracle.md:91]: Encoding types off here as well, I think

File: types/GasPriceOracle.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+
+## Dispute Considerations
+
+Note that following this guide does not prevent you from being disputed or guarantee reporters will properly put a value on-chain. Tellor is decentralized.  This repo is a start to the education necessary for a fully decentralized oracle, but please focus on communication and working with reporters to prevent unneccesary disputes and at the same time encourage monitoring and punishment of bad data. 
+
+- retreiving a gas price from one historical transaction is not recommended. gas price is very volatile
+- retreiving the current gas price from metamask is not recommended. gas price is very volatile
+- care should be used when reporting from black box aggregators
+- use EIP-1559 gas strategy
```
Comment (oraclown) [inline types/GasPriceOracle.md:110]: Add link?

File: types/GasPriceOracle.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+
+
+## Suggested Data Sources
+
+Where can reporters retrieve the query's response? Are there APIs available so reporters can programmatically retrieve the data for your query's expected response? Are there multiple possible sources for the expected response?
+
+Owlracle privode's a "gas price history" API. It supports the following networks:
```
Comment (oraclown) [inline types/GasPriceOracle.md:118]: Typos. *The GasPriceOracle query type provides

File: types/GasPriceOracle.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
+This is an example template for a new Query type. Every `Query` specification must include the following:
+
+
+## Type Name
+
+`Gas Price Oracle`
```
Comment (oraclown) [inline types/GasPriceOracle.md:6]: Should be `GasPriceOracle`

File: types/GasPriceOracle.md (hunk 2 above)
Comment (tallywiesenberg) [reply types/GasPriceOracle.md:45]: good catch

File: types/GasPriceOracle.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+`0x7dd07b75e8c096ea1926e9a2cc8f749b04581d0ae88a892989f05790384344f6`
+
+You can use [this tool](https://queryidbuilder.herokuapp.com/custom) to generate query IDs.
+
+
+## Dispute Considerations
+
+Note that following this guide does not prevent you from being disputed or guarantee reporters will properly put a value on-chain. Tellor is decentralized.  This repo is a start to the education necessary for a fully decentralized oracle, but please focus on communication and working with reporters to prevent unneccesary disputes and at the same time encourage monitoring and punishment of bad data. 
+
+- retreiving a gas price from one historical transaction is not recommended. gas price is very volatile
```
[the remark this answers is not in the document]
Comment (tallywiesenberg) [reply types/GasPriceOracle.md:107]: good point. also added some additional instructions for clarity

File: types/GasPriceOracle.md (hunk 4 above)
Comment (tallywiesenberg) [reply types/GasPriceOracle.md:114]: not sure what you mean

File: types/GasPriceOracle.md (hunk 5 above)
Comment (tallywiesenberg) [reply types/GasPriceOracle.md]: fixed the typo

File: types/GasPriceOracle.md (hunk 6 above)
Comment (tallywiesenberg) [reply types/GasPriceOracle.md]: true, fixed

File: types/GasPriceOracle.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+keccak256(queryData)
+```
+
+You can use [this tool](https://queryidbuilder.herokuapp.com/custom) to generate query IDs.
+
+
+## JSON Representation
+The JSON representation of your new query type is needed to construct query objects in a variety of languages. It contains the essential components of your query: type name, parameters in an ordered list and their corresponding value types, as well as the expected response type for the query.
+
+For example, the JSON representation of a `SpotPrice` query:
```
Comment (oraclown) [inline types/GasPriceOracle.md:63]: Says `SpotPrice`. Remove that part as this is `GasPriceOracle`

**Review by oraclown — APPROVED** [review]
Just one more specific fix requested and then maybe overall remove some of the default explanations of each section. And instead just point to the new query type template if the user needs more info about what a query is and its different components.

Looks good otherwise! Nice addition @tallywiesenberg !

**Pull request merged**